### ISSUE TITLE
Health of Curiefense incubation project

### ISSUE THREAD

**Issue by emaildanwilson** [cncf/toc#1192]
Is Curiefense being actively maintained?

I was looking for a new release that includes HTTP2 Rapid Reset fixed versions of their upstream proxy dependencies but see there have not been any recent published releases. I ended up needing to build my own but this really got me curious about the project.

What I've found so far.

Code updates seem to have stalled...

* No new merged PR's since July
* No new commits to the main repo since July

<img width="872" alt="Screen Shot 2023-10-17 at 8 52 10 AM" src="https://github.com/cncf/toc/assets/3026995/6a70e88d-ebca-4ee6-9c7d-902da87fd462">

Slack channels have gone dark

<img width="1118" alt="Screen Shot 2023-10-17 at 8 50 22 AM" src="https://github.com/cncf/toc/assets/3026995/b799fa99-9016-4a4a-b5af-6266545886a3">

I've been unable to get a response from any of the top contributors.

According to the last annual review it was still 100% maintained by Reblaze and they were hopeful that adoption would pick up and then they could work on diversifying contributions. No review was submitted for this year which could be because the review process is being deprecated.

**Comment by emaildanwilson** [comment]
Adding some metrics from https://all.devstats.cncf.io/d/54/project-health-table?orgId=1&var-repogroup_name=Curiefense

PRs: Number of PRs merged in the last 3 months: 0
Issues: Number of issues closed in the last 6 months: 0

These metrics do show commits over the last 3 month but they appear to be from a fork that is auto synced and not from any actual projects at https://github.com/curiefense.

cc  CODEOWNERS @tzuryby @xavier-rbz @phil777 @yitzchake @Aviv-Galmidi @flaper87 @jdorfman

**Comment by TheFoxAtWork** [comment]
@amye will reach out to the project to understand the current status

**Comment by amye** [comment]
Based on the responses in the issue with the project, this should likely move to archive due to inactivity.

**Comment by TheFoxAtWork** [comment]
Let's move this to a vote for Archive for the TOC @amye .  Do we need a separate issue/PR to conduct the vote or will this one suffice?

reference comment from former Project maintainer: https://github.com/curiefense/curiefense/issues/1259#issuecomment-1920812155

**Comment by amye** [comment]
I have marked this as 'ready for TOC', when we have a fully seated TOC and projects are moving to a vote, this will be in the queues.

**Comment by emaildanwilson** [comment]
We (Control Plane Corporation) are willing to provide engineering resources to begin maintaining the repo and garner additional participation from others in the CNCF community. Is it possible for us to do this?

Thank you for consideration!

**Comment by TheFoxAtWork** [comment]
I would say yes it is very possible. Couple thoughts (@jeefy if you could chime in on the details for how to go about the maintainer changes)
- we'd need to identify new maintainers to add. We'll need to have demonstrated effort/attempts to reincorporate the previous maintainers or receive notice of their divestment and withdrawal from the project. if notice is not received in a specified, realistic timeframe, we should annotate those attempts that show reasonable attempt before removal. this should be done transparently and documented. (we have this from 1 maintainer already and therefore referencing that previous comment should be done upon removal)
- project governance will need updated.
- project roadmap will need re-evaluated and updated

**Comment by TheFoxAtWork** [comment]
@tzuryby - given the interest by @emaildanwilson and Control Plan Corporation, do you have a defined path for onboarding new maintainers to the project? I'd like to see this project return or embark on the path to health with a full roadmap in the next 4 months.

**Comment by TheFoxAtWork** [comment]
@emaildanwilson  - I appreciate the support for the project. The TOC is going to discuss this internally to determine a best path forward. Archival is still part of that discussion but would not prevent Control Plane from forking the project and beginning development until we (TOC) reach consensus on direction. This would provide the most flexibility for any organization looking to pick up the project, regardless if it is determined that archival is a good path or not.

**Comment by dims** [comment]
Thank you, @jvassev @MajidAbuRmila @Cuppojoe @emaildanwilson, for your generous offer to take over the leadership and development of the Curiefense project. 

One of the considerations that has been raised in the past, regarding the potential transition of project ownership, is the concern over the name, brand, and trademark associations, as well as the overall vision and direction for the project. This is particularly relevant in the event that some of the previous committers and contributors decide to rejoin the project at a later stage. To address this concern and ensure a clean and organized transition, the most appropriate approach would be to mark the current Curiefense project [1][2] as archived.

By officially archiving the existing project, we can clearly communicate to the end-users and the broader community that they should no longer rely on the current project or expect further updates or support. Additionally, archiving the project would effectively remove its current "Incubation" status within the Cloud Native Computing Foundation (CNCF) ecosystem.

Once the archiving process is complete, you and your team would be free to embark on a fresh start with a new project. This would involve forking the existing Curiefense repository and creating a new, independent project with its own branding, vision, and governance structure. Furthermore, you would be able to file a new sandbox application with the CNCF, allowing your project to go through the established process for incubation and potential graduation as a CNCF project in its own right.

By taking this approach, you would have the freedom to shape the project according to your collective vision, without the potential constraints or obligations associated with the existing Curiefense project and its history. At the same time, this would provide a clear delineation between the previous project and your new endeavor, ensuring transparency and avoiding any confusion or conflicts down the line.

We understand that this may involve additional work and effort on your part, but we believe it is the most prudent and responsible way to proceed, ensuring a smooth transition and setting your project up for long-term success.

[1] https://www.cncf.io/archived-projects/ 
[2] https://github.com/cncf/toc/blob/main/process/archiving.md

**Comment by mrbobbytables** [comment]
Following up on @dims previous response, the TOC has decided to move this to a vote for archival.

/vote

**Comment by ctenberge-oneid** [comment]
Does anyone know if the project was resurrected with a different name?

**Comment by TheFoxAtWork** [comment]
@emaildanwilson See @ctenberge-oneid 's comment above. I dont recall hearing of a path forward once the project was archived, was this picked up by Control Plane Corporation as a fork?

**Comment by zuriby** [comment]
hello everyone. project creator here.
Reblaze, the company I founded back in 2011 was funding the development of this project.
Company got sold toward the end of year of 2023, and one of their first decisions were made was to shut this down.
By agreement I am prevented "competing" with them for a while, hence could not continue on my own.
That is unfortunate. You may archive it for good.

**Comment by emaildanwilson** [comment]
Control Plane is currently maintaining an internal fork and may consider switching it to a public project if there is enough demand and possible contributors.

**Comment by zuriby** [comment]
That would be awesome. I'll be happy to see this project back to life after it got shut down by the new owners of the company. I put so much into it, and it was devastating seeing it going down overnight.

**Comment by MrCzz** [comment]
@emaildanwilson Thanks for sharing! Nice to hear you’ve got this running as an internal fork and might make it public. That would definitely be great for the community. Have you already decided if you’re going to open source it?
